### ISSUE TITLE
RangeError: Source is too large

### ISSUE THREAD

**Issue by malinga91** [jackedgson/crunker#4]
I tried to add 15 small (max length 5sec) .mp3 files. After that I got following error. How to solve this error?

```
Crunker.js:38 Uncaught (in promise) RangeError: Source is too large
    at Float32Array.set (<anonymous>)
    at buffers.map.buffer (Crunker.js:38)
    at Array.map (<anonymous>)
    at Crunker.concatAudio (Crunker.js:37)
    at eval (eval at <anonymous> (bundle.min.js:44), <anonymous>:60:30)
```
<img width="722" alt="screen shot 2018-03-22 at 12 04 13 pm" src="https://user-images.githubusercontent.com/7546904/37754834-2c926866-2dc9-11e8-9c40-1e1325c7b325.png">

**Comment by jaggad** [comment]
Did you manage to resolve this issue?

**Comment by malinga91** [comment]
@jackedgson still not... :( postponed the development. 
Do you have a solution? 

Actually i need to build something like http://sampulator.com/. everything is done except joining clips together. 

Still this is a great plugin 👍 💯

**Comment by jaggad** [comment]
Hmm, I've not seen this before.

Can you please try pulling in a static mp3 (try a song) or just find a [random audio file](https://freesound.org/data/previews/423/423426_537412-lq.mp3) and testing that it works on you environment.

My first thought is the format for the audio file is not correct. This is likely an issue with a malformed `mp3` or just a type of mp3 didn't test.

**Comment by rafaelbusetti** [comment]
Same here.

I tested with the following files:
a) http://www.auxteam.com.br/merge-audio/1.mp3 
b)http://www.auxteam.com.br/merge-audio/2.mp3

Function "mergeAudio" work great, but concatAudio return this error

RangeError: Source is too large
    at Float32Array.set (<anonymous>)
    at crunker.js:61
    at Array.map (<anonymous>)
    at Crunker.concatAudio (crunker.js:56)
    at audio.fetchAudio.then.buffers ((index):18)

**Comment by Alynva** [comment]
After hours testing merge and concatenat functions, I found that simple lib. But my output file keep with the same stranger effect. For the simplicity of this lib, I decided to look into the code. Looking this issue, of concatenation, I saw that the output buffer was with 350207 of length for two samples with 190589 of length, what shuld result in a buffer with 381178 of length. Other thing that I observe, is that the sampleRate of my sample was 48k, the rate of the buffer was 44.1k and the length of the buffer is defined by the total duration os the samples multiplied by 44.1k. Changing the 44.1k to 48k, voy a la: worked!

I know that 44.1k is default, but I have more than 200 samples and for me, is more convenient keep the sources and adapt the code.

A possible soluction is use the sampleRate provide from the source (dynamic). Or just change the method to define the length of buffer (insted of use `_totalDuration()*sampleRate`, use just `_totalLength()`). But, for now, I can't work in it to make a PR.

Oh, and the README code has little bugs...

**Comment by jaggad** [comment]
@Alynva Just looked over your fork, seems to check out and makes sense. Your edits should definitely be merged into the master repository, please create a pull request to do so.

On a side note, it might even be worth [auto detecting the sampleRate](https://developer.mozilla.org/en-US/docs/Web/API/BaseAudioContext/sampleRate) of the input buffers. Of course there is the issue of two files with different sample rates being merged/concatenated.

Could be worth adding a brief note about sample rates and issues to the `README`.

**Comment by jaggad** [comment]
Fixed by #6
